### PR TITLE
fix: prevent selecting duplicate files in the same modal session

### PR DESCRIPTION
**Pull request by andrewleith**
# Summary | Résumé
This pull request improves the file attachment modal by preventing users from selecting the same file more than once in a single session. It also adds a test to verify that duplicate files are detected and reported as issues.

# Test instructions | Instructions pour tester la modification
**Prevents duplicates between reloads**
- Go to an email template, click attach more files
- Click choose files, select a file and attach it
- Reload the page
- Attach the same file
- [x] An error message is triggered and the file is not attached

**Prevents duplicates done one at a time**
- Go to an email template, click attach more files
- Click choose files, select a file and attach it
- Click attach more files again
- Select the same file and attach it again
- [x] An error message is triggered and the file is not attached

**Prevents duplicates done at the same time**
- Go to an email template, click attach more files
- Click choose files, select a file and attach it
- Click choose files, select the same file and attach it again
- [ ] An error message is triggered and the file is not attached

### CODE REVIEW

File: app/assets/javascripts/attachments/AttachFilesModal.js [hunk 1]
```diff
@@ -152,11 +152,24 @@ export const AttachFilesModal = ({
...
 
+    // Check for duplicates against files already selected in this modal session
+    const updatedIssues = [...validation.issues];
+    const filteredAcceptedFiles = validation.acceptedFiles.filter((file) => {
+      const isDuplicateInSession = selectedFiles.some(
+        (pendingFile) => pendingFile.file.name === file.name,
+      );
+      if (isDuplicateInSession) {
+        updatedIssues.push(copy.duplicateFilename(file.name));
+        return false;
```
[the remark this answers is not in the document]
Comment (andrewleith) [reply app/assets/javascripts/attachments/AttachFilesModal.js:163]: Its fine.

**Review by jzbahrai — APPROVED** [review]
